marketplace/app/v0/object_storage.py

```
@@ -0,0 +1,204 @@
+import json
+
+import marketplace_standard_app_api.models.object_storage as object_storage
+from fastapi import UploadFile
+
+from ..utils import check_capability_availability
+from .base import _MarketPlaceAppBase
+from .utils import _decode_metadata, _encode_metadata
+
+
+class MarketPlaceObjectStorageApp(_MarketPlaceAppBase):
+    @check_capability_availability
+    def list_collections(
+        self, limit: int = 100, offset: int = 0
+    ) -> object_storage.CollectionListResponse:
+        return object_storage.CollectionListResponse(
+            **self._client.get(
+                "listCollections", params={"limit": limit, "offset": offset}
+            ).json()
+        )
+
+    @check_capability_availability
+    def list_datasets(
+        self,
+        collection_name: object_storage.CollectionName,
+        limit: int = 100,
+        offset: int = 0,
+    ) -> object_storage.DatasetListResponse:
+        return object_storage.DatasetListResponse(
+            **self._client.get(
+                "listDatasets",
+                params={
+                    "collection_name": collection_name,
+                    "limit": limit,
+                    "offset": offset,
+                },
+            ).json()
+        )
+
+    @check_capability_availability
+    def create_or_update_collection(
+        self,
+        metadata: dict = None,
+        collection_name: object_storage.CollectionName = None,
+    ):
+        return self._client.put(
+            "createOrUpdateCollection",
+            params={"collection_name": collection_name} if collection_name else {},
+            headers=_encode_metadata(metadata),
+        ).json()
+
+    @check_capability_availability
+    def delete_collection(self, collection_name: object_storage.CollectionName):
+        return self._client.delete(
+            "deleteCollection", params={"collection_name": collection_name}
+        ).json()
```

Typically delete operation do not need to return anything. They either succeed or fail. In the latter case we need to raise an exception.